pkg/network/protocols/http/protocol.go

```diff
@@ -0,0 +1,215 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+// +build linux_bpf
+
+package http
+
+import (
+	"fmt"
+	"strings"
+	"unsafe"
+
+	"github.com/cilium/ebpf"
+	"github.com/davecgh/go-spew/spew"
+
+	manager "github.com/DataDog/ebpf-manager"
+
+	ddebpf "github.com/DataDog/datadog-agent/pkg/ebpf"
+	"github.com/DataDog/datadog-agent/pkg/network/config"
+	netebpf "github.com/DataDog/datadog-agent/pkg/network/ebpf"
+	"github.com/DataDog/datadog-agent/pkg/network/protocols"
+	"github.com/DataDog/datadog-agent/pkg/network/protocols/events"
+	"github.com/DataDog/datadog-agent/pkg/util/kernel"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+type httpProtocol struct {
+	cfg            *config.Config
+	telemetry      *Telemetry
+	statkeeper     *HttpStatKeeper
+	mapCleaner     *ddebpf.MapCleaner
+	eventsConsumer *events.Consumer
+}
+
+const (
+	httpInFlightMap = "http_in_flight"
+)
+
+func newHttpProtocol(cfg *config.Config) (protocols.Protocol, error) {
+	if !cfg.EnableHTTPMonitoring {
+		return nil, nil
+	}
+
+	kversion, err := kernel.HostVersion()
+	if err != nil {
+		return nil, fmt.Errorf("couldn't determine current kernel version: %w", err)
+	}
+
+	if kversion < MinimumKernelVersion {
+		return nil, fmt.Errorf("http feature not available on pre %s kernels", MinimumKernelVersion.String())
+
+	}
+
+	telemetry := NewTelemetry()
+	statkeeper := NewHTTPStatkeeper(cfg, telemetry)
+
+	return &httpProtocol{
+		cfg:        cfg,
+		telemetry:  telemetry,
+		statkeeper: statkeeper,
+	}, nil
+}
+
+func (p *httpProtocol) ConfigureOptions(mgr *manager.Manager, opts *manager.Options) {
+	if p == nil {
+		return
+	}
```

Comment:
It shouldn't happen, and I was hesitant to add this clause. My understanding is that it is a good practice to do so when using pointer receivers, but maybe we should remove it and let it panic if it goes further, as this would definitely be a programmer error from the user of the interface.

Reply:
So regarding pointer receivers, there is a best practice to use the same receiver type in a single struct implementation (meaning, we should not mix between value and pointer receivers).
We tend to always use pointer receivers to ensure we can have "editor" access to the instance.
For me, that check indicates that our code allows nil implementation of structs, and that's something we should eliminate. If nil implementation does nothing, we are just wasting cpu and memory on it.